src/kOS/Suffixed/StageValues.cs

```diff
@@ -56,9 +57,32 @@ private bool IsResource(string suffixName)
 
         private double? GetResourceOfCurrentStage(string resourceName)
         {
-            var resource = shared.Vessel.GetActiveResources();
-            var match = resource.FirstOrDefault(r => string.Equals(r.info.name, resourceName, StringComparison.InvariantCultureIgnoreCase));
-            return match == null ? null : (double?) Math.Round(match.amount, 2);
+            PartResourceDefinition resourceDef = PartResourceLibrary.Instance.resourceDefinitions.FirstOrDefault(pr => string.Equals(pr.name, resourceName, StringComparison.CurrentCultureIgnoreCase));
+            List<PartResource> list = new List<PartResource>();
+            if (resourceDef.resourceFlowMode == ResourceFlowMode.STACK_PRIORITY_SEARCH)
+            {
+                var engines = VesselUtils.GetListOfActivatedEngines(shared.Vessel);
+                foreach (var engine in engines)
+                {
+                    engine.GetConnectedResources(resourceDef.id, resourceDef.resourceFlowMode, list);
+                }
+            }
+            else
+            {
+                shared.Vessel.rootPart.GetConnectedResources(resourceDef.id, resourceDef.resourceFlowMode, list);
+            }
+            if (list.Count == 0)
+            {
+                return null;
+            }
+            double available = 0.0;
+            double capacity = 0.0;
+            foreach (var resource in list)
+            {
+                available += resource.amount;
+                capacity += resource.maxAmount;
```

Comment:
Yeah... I only did that while I was working on it because I thought it might be useful in the future.  There's no reason to be doing the extra math, and I'm almost always in favor of more LINQ.  Want me to make the change?